File: FootScout-Vue.WebClient/src/models/interfaces/Message.ts
import type { UserDTO } from '../dtos/UserDTO';
import type { Chat } from './Chat';

// Definiuje wiadomość w czacie, przechowując treść, nadawcę, odbiorcę i znacznik czasu.

export interface Message {
    id: number; 
    chatId: number;
    chat: Chat;
    content: string;
    senderId: string;
    sender: UserDTO;
    receiverId: string;
    receiver: UserDTO;
    timestamp: string;
}

File: FootScout-Vue.WebAPI/Entities/Message.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace FootScout_Vue.WebAPI.Entities
{
    // Model (encja) wiadomości
    public class Message
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int ChatId { get; set; }

        [ForeignKey("ChatId")]
        public virtual Chat Chat { get; set; }

        [Required]
        [MaxLength(200)]
        public string Content { get; set; }

        [Required]
        public string SenderId { get; set; }

        [ForeignKey("SenderId")]
        public virtual User Sender { get; set; }

        [Required]
        public string ReceiverId { get; set; }

        [ForeignKey("ReceiverId")]
        public virtual User Receiver { get; set; }

        [Required]
        [DataType(DataType.DateTime)]
        public DateTime Timestamp { get; set; }
    }
}